todos-app/src/App/App.tsx

```diff
@@ -1,18 +1,49 @@
-import React, { useState } from "react";
+import { useEffect, useState } from "react";
 import NewTodoInput from "./NewTodoInput/NewTodoInput";
 import TodoList from "./TodoList/TodoList";
 import ListFooter from "./ListFooter/ListFooter";
-import { ITodo } from "../DataStructure";
+import { Todo } from "./models/Todos";
+import { FilterType } from "./models/Filters";
 import "../App.css";
 
-const App: React.FC = () => {
-  const [todos, setTodos] = useState<ITodo[]>([]);
+const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterType>("All");
 
-  const addHandler = (input: ITodo) => {
+  useEffect(() => {
+    setFilterTodos(todos);
+    switch (filter) {
+      case "All": {
+        break;
+      }
+      case "Active": {
+        setFilterTodos((prev) =>
```

Comment:
Could this be a bit simpler, if you remove line 15 and instead here: `setFilterTodos(todos.filter(...))`

Reply:
If I understood everything correctly, then the app will not work correctly and the input will not displayed under the input